File: Basic Blockchain Structure/Blockchain with Python/blockchain.py
# Importing essential libraries
import datetime

# Class that shapes a blockchain object
class Blockchain:
  
  # The chain starts with the genesis block (first of all blocks, or root block),
  # with nonce 1 and root hash '0', by default.
  def __init__(self):
    self.chain = []
    self.create_block(nonce = 1, previous_hash = '0')

  def create_block(self, nonce, previous_hash, data):
    block = {'index': len(self.chain) + 1,
             'timestamp': str(datetime.datetime.now()),
             'nonce': nonce,
             'data': data,
             'previous_hash': previous_hash}
    self.chain.append(block)
    return block

File: Basic Blockchain Structure/Blockchain with Java/blockchain/Blockchain.java
package blockchain;

import java.util.ArrayList;
import java.util.Date;

public class Blockchain {
  
    private ArrayList<Block> chain;
    
    public Blockchain () {
        chain = new ArrayList<Block>();
        addBlock (1, "0", "0");
    }
  
    public ArrayList<Block> getChain() {
        return chain;
    }
    
    public Block CreateBlock (int nonce, String previousHash, String data) {
        Block block = new Block ( chain.size() + 1,
                                  new Date().toString(),
                                  nonce,
                                  data,
                                  previousHash );
        return block;
    }

    public void addBlock (Block block) {
        chain.add(block);
    }
    
    public void addBlock (int nonce, String previousHash, String data) {
        addBlock(CreateBlock(nonce, previousHash, data));
    }
}